Flaky test: TestAutoCloseIdleConnection

See [example failure](https://github.com/apache/pulsar-client-go/actions/runs/4349818338/jobs/7599905344)


<details>
<summary>Logs</summary>
<pre><code>
=== RUN   TestAutoCloseIdleConnection
time="2023-03-07T01:58:28Z" level=info msg="Connecting to broker" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:28Z" level=info msg="TCP connection established" local_addr="127.0.0.1:57540" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:28Z" level=info msg="Connection is ready" local_addr="127.0.0.1:57540" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:28Z" level=info msg="Connected consumer" consumerID=1 name=mvdoi subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Created consumer" consumerID=1 name=mvdoi subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Connected producer" cnx="127.0.0.1:57540 -> 127.0.0.1:6650" epoch=0 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Created producer" cnx="127.0.0.1:57540 -> 127.0.0.1:6650" producerID=1 producer_name=standalone-0-59 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Closing producer" producerID=1 producer_name=standalone-0-59 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Closed producer" producerID=1 producer_name=standalone-0-59 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="close producer, exit reconnect" producerID=1 producer_name=standalone-0-59 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Closing consumer=1" consumerID=1 name=mvdoi subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="Closed consumer" consumerID=1 name=mvdoi subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:28Z" level=info msg="close consumer, exit reconnect" consumerID=1 name=mvdoi subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Connecting to broker" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:34Z" level=info msg="TCP connection established" local_addr="127.0.0.1:52114" remote_addr="pulsar://localhost:6650"
==================
WARNING: DATA RACE
Read at 0x00c0003bb298 by goroutine 111:
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).checkAndCleanIdleConnections()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:157 +0x24d
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask.func1()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0x47

Previous write at 0x00c0003bb298 by goroutine 69:
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).connect()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:277 +0x5d2
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start.func1()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:230 +0x34

Goroutine 111 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0xd7
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:[1213](https://github.com/apache/pulsar-client-go/actions/runs/4349818338/jobs/7599905344#step:5:1214) +0x3d7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:229 +0xae
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).GetConnection()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:115 +0x67e
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).Request()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:120 +0xee
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).RequestToAnyBroker()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:103 +0x198
  github.com/apache/pulsar-client-go/pulsar/internal.(*lookupService).GetPartitionedTopicMetadata()
      /pulsar/pulsar-client-go/pulsar/internal/lookup_service.go:215 +0x283
  github.com/apache/pulsar-client-go/pulsar.(*client).TopicPartitions()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:238 +0xcd
  github.com/apache/pulsar-client-go/pulsar.(*consumer).internalTopicSubscribeToPartitions()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:303 +0xa7
  github.com/apache/pulsar-client-go/pulsar.newInternalConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:255 +0x5d4
  github.com/apache/pulsar-client-go/pulsar.newConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:195 +0x126a
  github.com/apache/pulsar-client-go/pulsar.(*client).Subscribe()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:206 +0x84
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:1220 +0x4bc
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================
==================
WARNING: DATA RACE
Read at 0x00c000459e20 by goroutine 111:
  github.com/apache/pulsar-client-go/pulsar/log.(*logrusWrapper).Debugf()
      /pulsar/pulsar-client-go/pulsar/log/wrapper_logrus.go:79 +0x47
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).checkAndCleanIdleConnections()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:157 +0x279
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask.func1()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0x47

Previous write at 0x00c000459e20 by goroutine 69:
  github.com/apache/pulsar-client-go/pulsar/log.(*logrusWrapper).SubLogger()
      /pulsar/pulsar-client-go/pulsar/log/wrapper_logrus.go:40 +0x6e
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).connect()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:277 +0x5b5
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start.func1()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:230 +0x34

Goroutine 111 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0xd7
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:1213 +0x3d7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:229 +0xae
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).GetConnection()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:115 +0x67e
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).Request()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:120 +0xee
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).RequestToAnyBroker()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:103 +0x198
  github.com/apache/pulsar-client-go/pulsar/internal.(*lookupService).GetPartitionedTopicMetadata()
      /pulsar/pulsar-client-go/pulsar/internal/lookup_service.go:215 +0x283
  github.com/apache/pulsar-client-go/pulsar.(*client).TopicPartitions()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:238 +0xcd
  github.com/apache/pulsar-client-go/pulsar.(*consumer).internalTopicSubscribeToPartitions()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:303 +0xa7
  github.com/apache/pulsar-client-go/pulsar.newInternalConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:255 +0x5d4
  github.com/apache/pulsar-client-go/pulsar.newConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:195 +0x126a
  github.com/apache/pulsar-client-go/pulsar.(*client).Subscribe()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:206 +0x84
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:1220 +0x4bc
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================
==================
WARNING: DATA RACE
Read at 0x00c0002477a0 by goroutine 111:
  github.com/sirupsen/logrus.(*Entry).Logf()
      /pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:332 +0x50
  github.com/sirupsen/logrus.(*Entry).Debugf()
      /pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:342 +0x64
  github.com/apache/pulsar-client-go/pulsar/log.(*logrusWrapper).Debugf()
      /pulsar/pulsar-client-go/pulsar/log/wrapper_logrus.go:79 +0x75
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).checkAndCleanIdleConnections()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:157 +0x279
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask.func1()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0x47

Previous write at 0x00c0002477a0 by goroutine 69:
  github.com/sirupsen/logrus.(*Entry).WithFields()
      /pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:151 +0x56c
  github.com/apache/pulsar-client-go/pulsar/log.(*logrusWrapper).SubLogger()
      /pulsar/pulsar-client-go/pulsar/log/wrapper_logrus.go:40 +0x51
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).connect()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:277 +0x5b5
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start.func1()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:230 +0x34

Goroutine 111 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.StartCleanConnectionsTask()
      /pulsar/pulsar-client-go/pulsar/internal/helper.go:25 +0xd7
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:1213 +0x3d7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47

Goroutine 69 (running) created at:
  github.com/apache/pulsar-client-go/pulsar/internal.(*connection).start()
      /pulsar/pulsar-client-go/pulsar/internal/connection.go:229 +0xae
  github.com/apache/pulsar-client-go/pulsar/internal.(*connectionPool).GetConnection()
      /pulsar/pulsar-client-go/pulsar/internal/connection_pool.go:115 +0x67e
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).Request()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:120 +0xee
  github.com/apache/pulsar-client-go/pulsar/internal.(*rpcClient).RequestToAnyBroker()
      /pulsar/pulsar-client-go/pulsar/internal/rpc_client.go:103 +0x198
  github.com/apache/pulsar-client-go/pulsar/internal.(*lookupService).GetPartitionedTopicMetadata()
      /pulsar/pulsar-client-go/pulsar/internal/lookup_service.go:215 +0x283
  github.com/apache/pulsar-client-go/pulsar.(*client).TopicPartitions()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:238 +0xcd
  github.com/apache/pulsar-client-go/pulsar.(*consumer).internalTopicSubscribeToPartitions()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:303 +0xa7
  github.com/apache/pulsar-client-go/pulsar.newInternalConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:255 +0x5d4
  github.com/apache/pulsar-client-go/pulsar.newConsumer()
      /pulsar/pulsar-client-go/pulsar/consumer_impl.go:195 +0x126a
  github.com/apache/pulsar-client-go/pulsar.(*client).Subscribe()
      /pulsar/pulsar-client-go/pulsar/client_impl.go:206 +0x84
  github.com/apache/pulsar-client-go/pulsar.TestAutoCloseIdleConnection()
      /pulsar/pulsar-client-go/pulsar/client_impl_test.go:1220 +0x4bc
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1486 +0x47
==================
time="2023-03-07T01:58:34Z" level=warning msg="Failed to perform initial handshake" error="unable to read frame size: connection closed" local_addr="127.0.0.1:52114" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:34Z" level=info msg="Connecting to broker" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:34Z" level=info msg="TCP connection established" local_addr="127.0.0.1:52124" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:34Z" level=info msg="Connection is ready" local_addr="127.0.0.1:52124" remote_addr="pulsar://localhost:6650"
time="2023-03-07T01:58:34Z" level=info msg="Connected consumer" consumerID=2 name=nabym subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Created consumer" consumerID=2 name=nabym subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Connected producer" cnx="127.0.0.1:52124 -> 127.0.0.1:6650" epoch=0 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Created producer" cnx="127.0.0.1:52124 -> 127.0.0.1:6650" producerID=2 producer_name=standalone-0-60 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Closing producer" producerID=2 producer_name=standalone-0-60 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Closed producer" producerID=2 producer_name=standalone-0-60 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="close producer, exit reconnect" producerID=2 producer_name=standalone-0-60 topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Closing consumer=2" consumerID=2 name=nabym subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="Closed consumer" consumerID=2 name=nabym subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
time="2023-03-07T01:58:34Z" level=info msg="close consumer, exit reconnect" consumerID=2 name=nabym subscription=my-sub topic="persistent://public/default/TestAutoCloseIdleConnection"
    testing.go:[1312](https://github.com/apache/pulsar-client-go/actions/runs/4349818338/jobs/7599905344#step:5:1313): race detected during execution of test
--- FAIL: TestAutoCloseIdleConnection (6.44s)
</code></pre>
</details>

--- a/pulsar/internal/connection_pool.go
+++ b/pulsar/internal/connection_pool.go
@@ -154,7 +154,8 @@ func (p *connectionPool) checkAndCleanIdleConnections(maxIdleTime time.Duration)
 			p.Lock()
 			for k, c := range p.connections {
 				if c.CheckIdle(maxIdleTime) {
-					c.log.Debugf("Closed connection due to inactivity.")
+					p.log.Debugf("Closed connection from pool due to inactivity. logical_addr=%+v physical_addr=%+v",
+						c.logicalAddr, c.physicalAddr)
 					delete(p.connections, k)
 					c.Close()
 				}

